### PR TITLE
Migrate Identity GC to use CEP and CES from hive.

### PR DESCRIPTION
**Pull request by alan-kut**
operator: Migrate Identity GC to use CEP and CES from hive.

```release-note
Operator Identity GC stops using legacy CEP and CES watchers.
```

### CODE REVIEW

**Comment by alan-kut** [comment]
It shares two commits with PR #26836

**Comment by alan-kut** [comment]
@pippolo84 FYI

File: operator/identitygc/gc_test.go [hunk 1]
```diff
@@ -5,39 +5,31 @@ package identitygc
...
 	v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
 	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
-	"github.com/cilium/cilium/pkg/k8s/informer"
-	"github.com/cilium/cilium/pkg/k8s/utils"
 	"github.com/cilium/cilium/pkg/option"
 )
 
 func TestIdentitiesGC(t *testing.T) {
-	defer goleak.VerifyNone(
+	/*defer goleak.VerifyNone(
```
Comment (joamaki) [inline operator/identitygc/gc_test.go]: Please try and fix the leaks and add this back (or add ignores if they're unfixable)

**Comment by alan-kut** [comment]
I'm including this into #26836

File: operator/identitygc/gc_test.go (hunk 1 above)
Comment (alan-kut) [reply operator/identitygc/gc_test.go]: Fixed and included in the #26836

**Pull request closed without merging**